### PR TITLE
test the archive function

### PR DESCRIPTION
**Pull request by smn**

### CODE REVIEW

File: vxpolls/example.py [hunk 1]
```diff
@@ -93,23 +93,25 @@ def on_message(self, participant, poll, message):
...
+            participant.has_unanswered_question = False
             yield self.reply_to(message, response, continue_session=False)
         else:
             response = config.get('survey_completed_response',
                                     self.survey_completed_response)
             yield self.reply_to(message, response, continue_session=False)
-            participant.poll_id = None
-            participant.set_poll_uid(None)
             if poll.repeatable:
                 # Archive for demo purposes so we can redial in and start over.
```
Comment (hodgestar) [inline vxpolls/example.py]: This comment should go because there are plenty of reasons to use this feature in production.

File: vxpolls/example.py [hunk 2]
```diff
@@ -93,23 +93,25 @@ def on_message(self, participant, poll, message):
...
     def end_session(self, participant, poll, message):
-        participant.interactions = 0
-        participant.has_unanswered_question = False
         next_question = poll.get_next_question(participant)
         config = yield self.pm.get_config(poll.poll_id)
         if next_question:
             response = config.get('batch_completed_response',
                                     self.batch_completed_response)
+            participant.interactions = 0
+            participant.has_unanswered_question = False
```
Comment (hodgestar) [inline vxpolls/example.py]: Did these get moved down so they're not cleared when `poll.get_next_question()` is called? Should these maybe not be done in `.get_next_question()`? That will require an audit of where get_next_question is called though.

File: vxpolls/example.py (hunk 1 above)
Comment (smn) [inline vxpolls/example.py]: Thanks

File: vxpolls/example.py (hunk 2 above)
Comment (smn) [inline vxpolls/example.py]: `get_next_question()` doesn't touch either of these variables. It was moved further down so that when we archive the participant, we archive it before any of the values are reset, otherwise the archived value wouldn't be accurate.

Comment (hodgestar) [inline vxpolls/example.py]: I'm just a bit worried by the poking around in the internals of participant. Maybe `participant` needs `.batch_completed()` and `.poll_completed()` functions?

**Comment by dmaclay** [comment]
I'm a little nervous about this since archiving is built on sessions and we know sessions are a bit boken
In theory the prefixing in session ,manager is totally separate though

**Comment by hodgestar** [comment]
:+1:

**Comment by smn** [comment]
@dmaclay this doesn't add any new features, it just adds tests for already existing features and fixes an edge case bug that existed in the code running in production (which no-one has come across yet so far). Is your concern a :-1: ? If it is then please let me know how you want it changed.

**Comment by hodgestar** [comment]
@smn @dmaclay Let's try land the session manager prefix fix and then this.

**Comment by smn** [comment]
Landing this since the session manager prefix has landed.
